### PR TITLE
Dbms/session 2

### PR DESCRIPTION
**Pull request by meta-Niraj-Gupta**

### CODE REVIEW

File: Database-Management-System/Session-2/Assignment-2/DBMSAssignment-2-part-1.sql [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+*/
+
+SHOW TABLES;
+
+/*
+command to remove Product table of the StoreFront database.
+*/
+SET FOREIGN_KEY_CHECKS = 0;
+DROP TABLE Product;
+
```
Comment (meta-simran-kapoor) [inline Database-Management-System/Session-2/Assignment-2/DBMSAssignment-2-part-1.sql]: Enable FOREIGN_KEY_CHECKS

File: Database-Management-System/Session-2/Assignment-2/DBMSAssignment-2-part-2.sql [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+Display the list of products which don't have any images.
+*/
+
+SELECT 
+    p.product_id, p.product_name
+FROM
+    product p
+WHERE
+    p.product_id NOT IN (SELECT 
+        product_id
```
Comment (meta-simran-kapoor) [inline Database-Management-System/Session-2/Assignment-2/DBMSAssignment-2-part-2.sql]: Could return 1 p id multiple times, use of DISTINCT could have been helpful

File: Database-Management-System/Session-2/Assignment-2/DBMSAssignment-2-part-2.sql [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+    from
+        product_image);
+        
+/*
+Display all Id, Title and Parent Category Title for all the Categories listed, 
+sorted by Parent Category Title and then Category Title. (If Category is top category 
+then Parent Category Title column should display “Top Category” as value.)
```
Comment (meta-simran-kapoor) [inline Database-Management-System/Session-2/Assignment-2/DBMSAssignment-2-part-2.sql]: Solve -  If Category is top category 
then Parent Category Title column should display “Top Category” as value.

File: Database-Management-System/Session-2/Assignment-2/DBMSAssignment-2-part-2.sql [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+    a.category_name AS 'Parent Category',
+    b.category_name AS 'Category Title'
+FROM
+    category a
+        INNER JOIN
+    category b ON a.category_id = b.parent_category
+ORDER BY a.category_name,b.category_name;
+
+/*
+Display Id, Title, Parent Category Title of all the leaf Categories
```
Comment (meta-simran-kapoor) [inline Database-Management-System/Session-2/Assignment-2/DBMSAssignment-2-part-2.sql]: Redo

File: Database-Management-System/Session-2/Assignment-2/DBMSAssignment-2-part-3.sql [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+FROM
+    product_order po
+        JOIN
+    order_detail pod ON po.order_id = pod.order_id
+GROUP BY pod.order_id
+ORDER BY po.date DESC
+LIMIT 50;
+
+/*
+Display 10 most expensive Orders.
```
Comment (meta-simran-kapoor) [inline Database-Management-System/Session-2/Assignment-2/DBMSAssignment-2-part-3.sql]: can be achieved in a less complex way

File: Database-Management-System/Session-2/Assignment-2/DBMSAssignment-2-part-3.sql [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+WHERE
+    order_id IN (SELECT 
+        detail_id
+    FROM
+        order_detail
+    WHERE
+        status = 'shipped') AND DATEDIFF(Now(), date) > 10;
+
+/*
+Display list of shoppers which haven't ordered anything since last month.
```
Comment (meta-simran-kapoor) [inline Database-Management-System/Session-2/Assignment-2/DBMSAssignment-2-part-3.sql]: redo

File: Database-Management-System/Session-2/Assignment-2/DBMSAssignment-2-part-3.sql [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+WHERE
+    DATEDIFF(Now(), order_time) < 15
+Order BY userId;
+
+/*
+Display list of order items which are in “shipped” state for particular Order Id (i.e.: 1020))
+*/
+
+SELECT 
+    order_id
```
Comment (meta-simran-kapoor) [inline Database-Management-System/Session-2/Assignment-2/DBMSAssignment-2-part-3.sql]: detail id

File: Database-Management-System/Session-2/Assignment-2/DBMSAssignment-2-part-3.sql [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+FROM
+    order_detail
+WHERE
+    order_id = 1 AND status = 'shipped';
+
+/*
+Display list of order items along with order placed date which fall between Rs 20 to Rs 50 price
```
Comment (meta-simran-kapoor) [inline Database-Management-System/Session-2/Assignment-2/DBMSAssignment-2-part-3.sql]: redo

File: Database-Management-System/Session-2/Assignment-2/DBMSAssignment-2-part-3.sql [hunk 9]
```diff
@@ -0,0 +1,121 @@
...
+FROM
+    order_detail Item
+        JOIN
+    product_order PO ON Item.order_id = PO.order_id
+WHERE
+    Item.price BETWEEN 100 AND 5000;
+
+
+/*
+Update first 20 Order items status to “Shipped” which are placed today
```
Comment (meta-simran-kapoor) [inline Database-Management-System/Session-2/Assignment-2/DBMSAssignment-2-part-3.sql]: limit?

File: Database-Management-System/Session-2/Assignment-2/DBMSAssignment-2-part-3.sql [hunk 10]
```diff
@@ -0,0 +1,121 @@
...
+*/
+
+UPDATE product_order_detail 
+SET status = 'SHIPPED'
+WHERE order_id IN (SELECT order_id FROM product_order WHERE DATE(order_date) = CURDATE());
```
Comment (meta-simran-kapoor) [inline Database-Management-System/Session-2/Assignment-2/DBMSAssignment-2-part-3.sql]: distinct?

**Review by ghost — APPROVED** [review]

**Pull request merged**